services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js

```diff
@@ -36,13 +41,134 @@ const propTypes = {
   touch: PropTypes.func.isRequired,
   submit: PropTypes.func.isRequired,
   formErrors: PropTypes.objectOf(PropTypes.string),
+  // eslint-disable-next-line react/no-unused-prop-types
+  formValues: PropTypes.objectOf(PropTypes.any).isRequired,
 };
 
 const defaultProps = {
   formErrors: {},
 };
 
+const StepForms = (props, mineName, primaryContact, state, next, prev, handleSaveData) => [
+  {
+    title: "Get Started",
+    content: <MajorMineApplicationGetStarted />,
+    buttons: [
+      <>
+        <Link to={routes.EDIT_PROJECT.dynamicRoute(props.project?.project_guid)}>
+          <Button
+            id="step1-cancel"
+            type="secondary"
+            style={{ marginRight: "15px" }}
+            onClick={() => {}}
+          >
+            Cancel
+          </Button>
+        </Link>
+        <Button id="step1-next" type="primary" onClick={() => next()}>
+          Next
+        </Button>
+      </>,
+    ],
+  },
+  {
+    title: "Create Submission",
+    content: (
+      <MajorMineApplicationForm
+        isEditMode={state.isEditMode}
+        initialValues={{
+          mine_name: mineName,
+          primary_contact: primaryContact,
+          ...props.project.major_mine_application,
+        }}
+        project={props.project}
+        majorMinesApplicationDocumentTypesHash={props.majorMinesApplicationDocumentTypesHash}
+      />
+    ),
+    buttons: [
+      <>
+        <LinkButton
+          style={{ marginRight: "20px" }}
+          onClick={(e) =>
+            handleSaveData(
+              e,
+              {
+                ...props.formValues,
+                documents: [
+                  ...(props.formValues?.primary || []),
+                  ...(props.formValues?.spatial || []),
+                  ...(props.formValues?.supporting || []),
+                ],
+                status_code: "DFT",
+              },
+              "Successfully saved a draft major mine application."
+            )
+          }
+          disabled={!props.formValues?.primary?.length > 0}
+          title="Save Draft"
+        >
+          Save Draft
+        </LinkButton>
+        <Button
+          id="step-back"
+          type="tertiary"
+          className="full-mobile"
+          style={{ marginRight: "20px" }}
+          onClick={() => prev()}
+        >
+          Back
+        </Button>
+        <Button
+          id="step2-next"
+          type="primary"
+          onClick={() => next()}
+          disabled={!props.formValues?.primary?.length > 0}
+        >
+          Review & Submit
```

Comment:
Should we be able to advance to Review & Submit before saving an initial draft? The existing navigation allows us to step 3 without saving a draft.